server/routes/index.js

```diff
@@ -1,24 +1,21 @@
 import express from 'express';
 import UserController from '../controllers/userController';
-import SchoolController from '../controllers/schoolController';
-import ResultsController from '../controllers/resultsController';
 import UserValidator from '../middlewares/userValidation';
-import AddUsersValidation from '../middlewares/addUsersValidation';
-import SchoolValidator from '../middlewares/schoolValidation';
-import AddResultsValidation from '../middlewares/addResultsValidation';
-import { checkUserIsLoggedIn, checkUserRole } from '../middlewares/checkUser';
+import adminPriviledgeRouter from './adminPriviledge';
+import teacherPrivilegeRouter from './teacherPriviledge';
 
 
 const router = express.Router();
 // default route
 router.get('/', (req, res) => res.send('Welcome to EduGate!'))
 
 router.get('/api/v1/users/logout', UserController.logout);
-router.get('/api/v1/users/:role', checkUserIsLoggedIn, checkUserRole, UserController.getUsers);
+
 router.post('/api/v1/users/login', UserValidator.validateUserLogin, UserController.login);
 router.post('/api/v1/users/signup', UserValidator.validateUserSignUp, UserValidator.checkExistingEmail, UserController.signUp);
-router.post('/api/v1/users/addusers', AddUsersValidation.validateAddUsers, UserController.addUsers);
-router.post('/api/v1/schools/create', SchoolValidator.validateSchoolCreate, SchoolController.create);
-router.post('/api/v1/results/addresults', AddResultsValidation.validateAddResults, ResultsController.addResults);
+
+router.use(adminPriviledgeRouter);
+router.use(teacherPrivilegeRouter)
+
```

Comment:
If we order it:

```
router.use(checkTeacherPrivilege);
router.use(teacherPrivilegeRouter);
router.use(checkAdminPrivilege);
router.use(adminPrivilegeRouter);
```

then it means we don't have to insert the middleware in each individual route within the router. 

What do you think?

Reply:
We should also create middleware that checks whether a user has a `school_uid` to make sure that they have registered their school after signing up, and if not, should redirect them to the school registration page.

Reply:
> If we order it:
> 
> ```
> router.use(checkUserIsLoggedIn);
> router.use(userRouter); // i.e dashboard, profile, settings etc.
> 
> router.use(checkTeacherPrivilege);
> router.use(teacherPrivilegeRouter);
> 
> router.use(checkAdminPrivilege);
> router.use(adminPrivilegeRouter);
> ```
> 
> then it means we don't have to insert the middleware in each individual route within the router.
> 
> What do you think?

Cool. It makes the codebase looks cleaner.

Reply:
@jamsgra I have implemented the changes. Pls have a look at it

Reply:
Hey so I think we have to be careful with the order.

Any middleware we use will be applied to everything below it.

So for instance with: 

```
router.use(checkAdminPrivilege);
router.use(adminPrivilegeRouter);

router.use(checkTeacherPrivilege);
router.use(teacherPrivilegeRouter);
```
The `checkAdminPrivilege` will be applied to the `teacherPrivilegeRouter` as it is executed before that router, and so this will prevent teachers accessing those routes as they will be blocked by the `checkAdminPrivilege` middleware.

Instead if we order it this way:
```
router.use(checkTeacherPrivilege);
router.use(teacherPrivilegeRouter);

router.use(checkAdminPrivilege);
router.use(adminPrivilegeRouter);
```

The `checkAdminPrivilege` will only be applied to the `adminPrivilegeRouter`

Reply:
@jamsgra checkAdminPrivilege before checkTeacherPrivilege because checkIfUserHasSchool should not apply to `create school` API endpoint and it lives in the `adminPrivilege.js`